core/src/main/java/com/segment/analytics/kotlin/core/Telemetry.kt

```diff
@@ -187,9 +189,13 @@ object Telemetry: Subscriber {
         if (!metric.startsWith(METRICS_BASE_TAG)) return
         if (tags.isEmpty()) return
         if (queue.size >= maxQueueSize) return
+        if (Math.random() > sampleRate) return
 
-        var filteredTags = tags.toMap()
-        if (!sendWriteKeyOnError) filteredTags = tags.filterKeys { it.lowercase() != "writekey" }
+        var filteredTags = if(sendWriteKeyOnError) {
```

Comment:
I really like Kotlin's alternative to ternary operators here.  Python's makes me a little nauseous  - `foo = something if condition else something_else` aaaaaugh